### PR TITLE
srcOnly: Some improvements

### PR DESCRIPTION
**Pull request by Pandapip1**
## Description of changes

Continuation of https://github.com/NixOS/nixpkgs/pull/344699

Take 2: Hopefully I don't rebase on the wrong branch this time!

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [x] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: nixos/doc/manual/release-notes/rl-2411.section.md [hunk 1]
```diff
@@ -38,6 +38,8 @@
   Users can use it by `services.displayManager.ly.enable` and config it by
   `services.displayManager.ly.settings` to generate `/etc/ly/config.ini`
 
+- `srcOnly` was rewritten to be more readable, have additional warnings in the event that something is probably wrong, use the `stdenv` provided by the derivation, and Noogle-compatible documentation was added.
```
Comment (philiptaron) [inline nixos/doc/manual/release-notes/rl-2411.section.md:41]: I'd drop the release notes, but I don't care enough to insist. Based on what's above and below this note, this is in some different class than NixOS module updates.

**Review by philiptaron — APPROVED** [review]

**Comment by trofi** [comment]
Bisect says 0b77f4ad4f3e5b3ed90cd95fc62d88c95910b414 `srcOnly: some readability improvements` broke my `nixos` system build as:

```
error: builder for '/nix/store/i5yqj8ii5rv166ns67iz0xbdn4frz7gm-glibc-2.40-36-source-unsecvars.drv' failed to produce output path for output 'bin' at '/nix/store/i5yqj8ii5rv166ns67iz0xbdn4frz7gm-glibc-2.40-36-source-unsecvars.drv.chroot/root/nix/store/irxl72hizfw8zz3wqyyj961ggrq6ivai-glibc-2.40-36-source-unsecvars-bin'
error: 1 dependencies of derivation '/nix/store/r5mk1mld0k6yghk3b0l4hs13halz3bqi-security-wrapper-chsh-x86_64-unknown-linux-musl.drv' failed to build
error: 1 dependencies of derivation '/nix/store/2pnq4zhv3hyns5agni7ycdva7fz3cmpp-unit-script-suid-sgid-wrappers-start.drv' failed to build
error: 1 dependencies of derivation '/nix/store/g9g3z2jgd7xddjyixsxg87c566wld86z-unit-suid-sgid-wrappers.service.drv' failed to build
error: 1 dependencies of derivation '/nix/store/vqs9wb7vfn5d2rqj5grcaxqjf681aq6p-system-units.drv' failed to build
error: 1 dependencies of derivation '/nix/store/irfa0dpz26al8dmckq2x8q58x4kqy6in-etc.drv' failed to build
error: 1 dependencies of derivation '/nix/store/mc5m0qhs0iip9i63vy7q1lbwrh58pb8q-nixos-system-nixos-24.11.20240929.0b77f4a.drv' failed to build
```

Probably because `bin` output is now still declared and expects some output.

**Comment by Pandapip1** [comment]
> Looks like outputs = [ "out" ]; was just lost. Probably because bin output is now still declared and expects some output.

Yup, clearly forgot to re-include that when rewriting. I'll submit a fix now.

**Comment by trofi** [comment]
> Have any derivations I can include as nixos tests?

As all wrappers failed in `nixos/modules/security/wrappers/default.nix` I would expect any existing test that exercises it to fail. Otherwise merely wrapping `glibc` should be enough:

```
# master: ok
$ nix build --impure --expr 'with import ./. {}; srcOnly glibc'

# staging (unfixed): fail
$ nix build --impure --expr 'with import ./. {}; srcOnly glibc'
error: builder for '/nix/store/vzhwaql5m22555lp017p289ybaiz5nns-glibc-2.40-36-source.drv' failed to produce output path for output 'bin' at '/nix/store/vzhwaql5m22555lp017p289ybaiz5nns-glibc-2.40-36-source.drv.chroot/root/nix/store/jw57377nvnn124wabps40kiqamnri1b2-glibc-2.40-36-source-bin'
```

**Pull request merged**